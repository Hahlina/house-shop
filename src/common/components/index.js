export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
