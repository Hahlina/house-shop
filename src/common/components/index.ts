export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
export * from './Container/Container';
export * from './Image/Image';
export * from './ErrorPage/ErrorPage';
export * from './SomethingWentWrong/SomethingWentWrong';
export * from './NothingFound/NothingFound';
