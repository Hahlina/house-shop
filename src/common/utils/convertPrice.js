export const convertPrice = (price) =>
    new Intl.NumberFormat('en-US', {
        style: 'currency',
        currency: 'USD'
    }).format(price || 0);
