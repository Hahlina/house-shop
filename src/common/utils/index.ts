export * from './convertPrice';
export * from './capitalizeFirstLetter';
