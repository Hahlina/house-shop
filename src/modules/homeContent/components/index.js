export * from "./PreviewBanner/PreviewBanner";
export * from "./PropertiesList/PropertiesList";
