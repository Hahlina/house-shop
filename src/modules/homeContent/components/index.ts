export * from './PreviewBanner/PreviewBanner';
export * from './PropertiesList/PropertiesList';
export * from './PropertiesPagination/PropertiesPagination';
