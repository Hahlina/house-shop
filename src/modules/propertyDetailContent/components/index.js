export * from "./PropertyGallery/PropertyGallery";
export * from "./FullInformationCard/FullInformationCard";
