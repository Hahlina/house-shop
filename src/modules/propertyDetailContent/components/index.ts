export * from './PropertyGallery/PropertyGallery';
export * from './PropertyGallery/PropertyGalleryLoader';
export * from './FullInformationCard/FullInformationCard';
export * from './FullInformationCard/FullInformationCardLoader';
export * from './PropertyDetailContentLoader/PropertyDetailContentLoader';
export * from './PropertyDetailBreadCrumbs/PropertyDetailBreadCrumbs';
