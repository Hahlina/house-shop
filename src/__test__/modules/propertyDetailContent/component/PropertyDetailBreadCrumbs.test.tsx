import { PropertyDetailBreadCrumbs } from 'modules/propertyDetailContent/components';

import { render, screen } from 'common/utils/test';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    Link: 'a',
    useParams: () => '1',
    useNavigate: jest.fn(),
    useLocation: () => () => ({ pathname: '/example/path' })
}));

describe('PropertyDetailBreadCrumbs component', () => {
    it('should render the component without crashing', () => {
        render(<PropertyDetailBreadCrumbs />);
    });

    it('should display a breadcrumb with a link to the home page', () => {
        render(<PropertyDetailBreadCrumbs />);
        expect(screen.getByText('Home')).toBeInTheDocument();
    });

    it('should display custom breadcrumbs Properties', () => {
        render(<PropertyDetailBreadCrumbs />);
        expect(screen.queryByText('Properties')).toBeInTheDocument();
    });

    it('should take a snapshot', () => {
        const { container } = render(<PropertyDetailBreadCrumbs />);
        expect(container).toMatchSnapshot();
    });
});
