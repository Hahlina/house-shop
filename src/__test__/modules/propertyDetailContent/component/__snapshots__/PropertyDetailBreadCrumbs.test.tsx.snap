// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyDetailBreadCrumbs component should take a snapshot 1`] = `
<div>
  <div
    class="sc-qZrbh bfdSXZ"
  >
    <nav
      aria-label="breadcrumbs"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            data-testid="breadcrumb-link"
            to="/"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 sc-hLQSwg fJWCBm css-ahj2mt-MuiTypography-root"
            >
              Home
            </p>
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            to="#"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 sc-hLQSwg fJWCBm css-ahj2mt-MuiTypography-root"
            >
              Properties
            </p>
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            to="/property/"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 sc-hLQSwg fJWCBm css-ahj2mt-MuiTypography-root"
            />
          </a>
        </li>
      </ol>
    </nav>
  </div>
</div>
`;
